html_app/components/MicroscopyModelFactory.js

```diff
@@ -67,45 +64,36 @@ scb.components.MicroscopyModelFactory = function scb_components_MicroscopyModelF
                                         collection_id = rule['use_collection_id'];
                                     }
                                     max = template.slide_parser[collection_id][slide_type][conditions].length;
-                                    index = Math.floor(Math.random() * (max - 1 + 1));
-									var slide_array = template.slide_parser[collection_id][slide_type][conditions][index];
-									var alreadySelected = false;
-									_.each(slide_array, function(x){if(_.contains(hash_list, x.hash)) alreadySelected=true;
-									});
-									var number_of_comparisons = 0;
-									while(alreadySelected && number_of_comparisons < max){
-										index =  Math.floor(Math.random() * (max - 1 + 1));
-										slide_array = template.slide_parser[collection_id][slide_type][conditions][index];
-										alreadySelected = false;
-										_.each(slide_array, function(x){if(_.contains(hash_list, x.hash)) alreadySelected=true;
-										});
-										number_of_comparisons=number_of_comparisons+1;
-									}
-									imgs=slide_array;
-									isFound = true;
-								}
-								else{
+                                    number_of_comparisons = 0;
+                                    do{
+                                        /*Math.random returns num in [0,1) so no need to worry about subtracting 1 from index*/
+                                        index =  Math.floor(Math.random() * (max));
+                                        slide_array = template.slide_parser[collection_id][slide_type][conditions][index];
+                                        alreadySelected = false;
+                                        _.each(slide_array, function(x){
+                                            if(_.contains(hash_list, x.hash))
```

Comment:
can you add braces here? I know they aren't strictly required for one line ifs, but it is much clearer if they are.